XFEM tests require --enable-unique-ids

jwpeterson opened the pull request:
This issue won't affect anyone that builds libmesh using the update_and_rebuild_libmesh.sh script, but I somehow neglected to ever add `--enable-unique-ids` to my libmesh build script, and the XFEM tests all require this to run correctly.  

Note that MOOSE and the modules still compile without unique ids enabled, and the rest of the tests run fine, so the failures were a bit surprising (and I sent @bwspenc on some wild goose chases looking for them) at first.

Refs #6453.

jwpeterson committed:
Add unique_id detection to TestHarness.

Refs #6453.

jwpeterson committed:
Flag xfem tests which require --enable-unique-id.

bwspenc commented:
This looks good, thanks!  The other piece we discussed is to put something like this in code that would always get run when you use XFEM:

```
#ifndef LIBMESH_ENABLE_UNIQUE_ID
mooseError("MOOSE requires unique ids to be enabled in libmesh (compile with --enable-unique-id) to use XFEM!");
#endif
```

I'd stick it in the constructor in ```modules/xfem/src/base/XFEM.C```.  Do you want to do that as part of this PR, or do that separately?

jwpeterson commented:
I'll add it to this one.

jwpeterson committed:
Throw mooseError in XFEM constructor if unique_ids not present.

This will prevent someone from running an XFEM app with an
incorrectly-configured libmesh and then getting a more confusing error
message later.

Refs #6453.

bwspenc commented:
:+1:

permcody commented on python/TestHarness/util.py:
```diff
@@ -75,6 +75,10 @@
+                     'default'   : 'FALSE',
+                     'options'   : {'TRUE' : '1', 'FALSE' : '0'}
+                   },
 }
```
For the record, the number of locations is five!

permcody commented:
:+1:
